helpers/compile/plugins/tscPlugin.ts

```diff
@@ -92,6 +92,17 @@ export const tscPlugin: (emitTypes?: boolean) => esbuild.Plugin = (emitTypes?: b
         if (options.incremental !== true && process.env.DEV !== 'true') {
           // we get the types generated by tsc and bundle them near the output
           bundleTypeDefinitions(typeOutPath, bundlePath)
+
+          // when types are already exported by us, it is not always guaranteed
+          // that the type bundler will also export them. It won't be able to
+          // trace them correctly back to runtime module and fail with either
+          // "... is using the type X but can not be named" or "The inferred
+          // type of this node exceeds the maximum length" error, or even worse
+          // the compiler could inline the types in the users' project causing
+          // very bad performance issues. To fix this, we export all the types.
```

Comment:
This is really really bad. This means that when it did not fail, TS was potentially inlining our non-exported types utils into user projects (given that they are using or re-exporting a dependent type). This, as I found out, and as was written in a previous comment could not only cause errors in extreme cases but also slow down compilation...